crates/formality-check/src/adts.rs

```diff
@@ -15,10 +18,17 @@ impl super::Check<'_> {
 
         self.prove_where_clauses_well_formed(&env, &where_clauses, &where_clauses)?;
 
-        // FIXME: check names are unique or integers from 0..n
-
-        for Variant { name: _, fields } in &variants {
-            for Field { name: _, ty } in fields {
+        // names is used to check that there are no name conflicts
+        let mut names = HashSet::new();
```

Comment:
can you move the name check above any trait solving? i.e. right after the `instantiate_universally`?

Reply:
I needed some way to get the term out of the `Binder`. I guess it doesn't matter how it's instantiated for the purposes of unique naming, so I chose `Binder::open()` but maybe there's a simpler way with eg `Binder::instantiate`?

Reply:
you can use the `Binder::peek` method, @shua